lib/fizzy/execute.cpp

```diff
@@ -840,7 +840,11 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                     throw std::bad_alloc();
                 memory->resize(new_pages * PageSize);
             }
-            catch (std::bad_alloc const&)
+            catch (const std::bad_alloc&)
+            {
+                ret = static_cast<uint32_t>(-1);
+            }
+            catch (const std::length_error&)
```

Comment:
Doesn't look possible to cover it, for me `std::string::max_size()` returns 2^46-1 bytes, we can't allocate more than that with wasm.

Reply:
I don't think this is worth having considering [`max_size()` is not very useful](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2012/n3440.html#197).
It may be just better to add a unit test that checks if wasm memory max size is not smaller than `std::string::max_size()`.
Alternatively, you can catch `std::exception` here.

Reply:
On 32-bit build the value is 2^31-1.